### PR TITLE
form validation for memo

### LINKED ISSUE #323: send dialog and plumbing does not handle over-long memos
1. generate a 512-byte memo plaintext, this is longer than the limit (the address and the text together are 512 bytes)
```
Python 3.9.6 (default, Nov 10 2023, 13:38:27)
[Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 'aaaaaaa ' * (512//8)
```
2. paste into memo field in send dialog
3. tx approval popup appears, click approve
4. tx build fails with inscrutable error message
Screenshot 2023-12-28 at 2 30 58 PM

two problems here:

1. form should be doing input validation on memo length
2. errors should be propagated back to user

### PR DESCRIPTION
**Pull request by turbocrime**
form validation for memo

### CODE REVIEW

File: apps/webapp/src/state/send.ts [hunk 1]
```diff
@@ -148,16 +149,19 @@ export const validateAmount = (asset: AssetBalance, amount: string): boolean =>
...
 export const sendValidationErrors = (
   asset: AssetBalance | undefined,
   amount: string,
   recipient: string,
+  memo?: string,
 ): SendValidationFields => {
   return {
     recipientErr: Boolean(recipient) && !isPenumbraAddr(recipient),
     amountErr: !asset ? false : validateAmount(asset, amount),
+    memoErr: new TextEncoder().encode(memo).length > 512 - BECH32_ADDRESS_LENGTH,
```
Comment (turbocrime) [inline apps/webapp/src/state/send.ts:164]: is this length calc correct

**Comment by turbocrime** [comment]
i only updated the validation function, but it might be nice to provide an indication of maximum length, or use an input attribute to actually limit the field.

**Comment by turbocrime** [comment]
error happens in wasm. so handler is failing to rethrow as ConnectError, and, not validating before calling wasm

**Comment by turbocrime** [comment]
the js shim provided by wasm-bundler is not rethrowing the error. this is the source of the error flow disconnect. the error received by the implementation is literally a webassembly 'unreachable'

we can try/catch around every wasm call, to address this. but a correct solution should involve fixing wasm crate to rethrow specific errors

File: apps/webapp/src/state/send.ts (hunk 1 above)
Comment (grod220) [reply apps/webapp/src/state/send.ts:164]: According to Henry's issue it's `address+memo <= 521 bytes`. So this looks right. Can we add a code comment right above?

```diff
+ // The memo plus the address cannot exceed 512 bytes
memoErr: new TextEncoder().encode(memo).length > 512 - BECH32_ADDRESS_LENGTH,
```

**Review by grod220 — APPROVED** [review]

**Comment by grod220** [comment]
Also, I'd change the PR description so it doesn't close the issue on merge because the Rust code still needs changed too.

File: apps/webapp/src/state/send.ts (hunk 1 above)
Comment (turbocrime) [reply apps/webapp/src/state/send.ts:150]: turns out this was incorrect, at least according to how wasm throws. it is using an 80-byte address in the memo and will throw between a 432-char -> 433-char memo length

**Pull request merged**